e2e/puppeteer_standalone.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+const puppeteer = require('puppeteer');
+const SERVER_URL = 'http://localhost:3001';
+
+const chromeOptions = {
+  executablePath:process.env.chrome,
+  headless: false
+};
+
+(async function main() {
+  try {
+    const browser = await puppeteer.launch(chromeOptions);
+    const page = await browser.newPage();
+    page.emulate({
+      viewport: {
+        width: 1280,
+        height: 1024
+      },
+      userAgent: ''
+    });
+    await page.goto(SERVER_URL, { waitUntil: 'domcontentloaded' });
+
+    // Run happy path test for submitting a transaction
+    await page.waitForSelector('#test_sender_component');
+    await page.waitForTimeout(1000);
+    await page.click('#test_sender_component').then(() => console.log('-- Open Sender dropdown'));
+    await page.waitForTimeout(1000);
+    const [aliceOption] = await page.$x("//div[contains(., '5sauUX')]");
+    await page.waitForTimeout(500);
+    if (aliceOption) {
+        await aliceOption.click().then(() => console.log('-- Click on an account from the dropdown'));
+    }
+    await page.waitForTimeout(500);
+    await page.focus('#test_amount_send').then(() => console.log('-- Focus on amount input field.'));
+    await page.keyboard.type('10').then(() => console.log('-- add 10 MLAU'));;
+    await page.waitForTimeout(500);
+    await page.focus('#receiver_input').then(() => console.log('-- Focus on receiver input field'));
+    await page.keyboard.type('74YBVK9EJ4uSzokqewqmBcecAWq6tosB2xujX5Lf8jKg2gze').then(() => console.log('-- Add a receiver address'));
+    await page.waitForTimeout(500);
+    await page.click('#test-button-submit').then(() => console.log('-- Click send button to initiate the transaction'));
+    await page.waitForXPath('//div[contains(text(), "Transaction was broadcasted")]').then(() => console.log('-- Transaction was broadcasted.'));
```

Comment:
Isn't this kind of selectors a bit too generic?

Reply:
Can you elaborate please?

Reply:
Same as with `alice` - like it can match anything on the page, even if it's invisible, hence I think it's a bit fragile.

We can maybe add a test-specific, invisible component which displays current transaction status and match on this? Obviously polluting production code with test code is not a very good practice, but from my experience it's much more practical than attempting to match exactly what the user sees - note the test here is meant to make sure that the transaction is being processed correctly, not necessarily that the right messages are being displayed.
The main problem in the past I had with E2E tests was that they are rarely reliable and people don't really know how to debug them, so minimizing flakiness is my priority here, hence proposing this separate test-specific component.

But we don't have to fix it now, just wanted to plant a seed. :)